src/models/WalletConnect.js

```diff
@@ -17,121 +17,132 @@
     with this program; if not, write to the Free Software Foundation, Inc.,
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
-// Based on types from:
-// https://github.com/WalletConnect/walletconnect-monorepo/blob/master/packages/types/index.d.ts
 
-export type ClientMeta = {|
-  description: string,
-  url: string,
-  icons?: string[],
-  name: string,
-|};
-
-export type Session = {|
-  connected: boolean,
-  accounts: string[],
-  chainId: number,
+export type WalletConnectConnector = {|
   bridge: string,
   key: string,
   clientId: string,
-  clientMeta?: ClientMeta,
   peerId: string,
-  peerMeta?: ClientMeta,
-  handshakeId: number,
+  clientMeta: WalletConnectClientMeta | null,
+  peerMeta: WalletConnectClientMeta | null,
   handshakeTopic: string,
+  handshakeId: number,
+  uri: string,
+  chainId: number,
+  networkId: number,
+  accounts: string[],
+  rpcUrl: string,
+  connected: boolean,
+  pending: boolean,
+  session: WalletConnectSession,
+  on(event: string, callback: (error: Error | null, payload: any | null) => void): void,
```

Comment:
For `Error` I prefer | null here since it's a copy of their typescript model (pity that it cannot be reused from lib tho) and it can be either provided model or null (as stated per type) while ? states it can also be undefined, wdyt?

Applied suggestion to payload tho 👍